static/src/javascripts/projects/facia/modules/onwards/weather.js

```diff
@@ -0,0 +1,212 @@
+        } else {
+            return this.getWeatherData(`${config.page.weatherapiurl}.json`)
+                .then(response => {
+                    this.fetchWeatherData(response);
+                })
+                .catch(err => {
+                    reportError(err, {
+                        feature: 'weather',
+                    });
+                });
+        }
+    },
+
+    fetchWeatherData(location: Location) {
+        return this.getWeatherData(
+            `${config.page.weatherapiurl}/${location.id}.json?_edition=${config.page.edition.toLowerCase()}`
+        )
+            .then(response => {
+                this.render(response, location.city);
+                this.fetchForecastData(location);
+            })
+            .catch(err => {
+                reportError(err, {
+                    feature: 'weather',
+                });
+            });
+    },
+
+    clearLocation() {
+        userPrefs.remove(prefName);
+        if (searchTool !== null) {
+            searchTool.setInputValue();
+        }
+    },
+
+    fetchForecastData(location: Location) {
+        return this.getWeatherData(
+            `${config.page.forecastsapiurl}/${location.id}.json?_edition=${config.page.edition.toLowerCase()}`
+        )
+            .then(response => {
+                this.renderForecast(response);
+            })
+            .catch(err => {
+                reportError(err, {
+                    feature: 'weather',
+                });
+            });
+    },
+
+    saveDeleteLocalStorage(response: CityPreference) {
+        // After user interaction we want to store the location in localStorage
+        if (response.store === 'set') {
+            this.saveUserLocation(response);
+            this.fetchWeatherData(response);
+
+            // After user sent empty data we want to remove location and get the default location
+        } else if (response.store === 'remove') {
+            this.clearLocation();
+            this.getDefaultLocation();
+        }
+    },
+
+    bindEvents() {
+        bean.on(document.body, 'click', '.js-toggle-forecast', e => {
+            e.preventDefault();
+            this.toggleForecast();
+        });
+
+        mediator.on(
+            'autocomplete:fetch',
+            this.saveDeleteLocalStorage.bind(this)
+        );
+    },
+
+    toggleForecast() {
+        $('.weather').toggleClass('is-expanded');
+    },
+
+    addSearch() {
+        searchTool = new SearchTool({
+            container: $('.js-search-tool'),
+            apiUrl: config.page.locationapiurl,
+        });
+    },
+
+    render(weatherData: WeatherResponse, city: string) {
+        this.attachToDOM(weatherData.html, city);
+
+        if (!eventsBound) {
+            this.bindEvents();
+            eventsBound = true;
+        }
+
+        if (searchTool === null) {
+            this.addSearch();
+        } else {
+            searchTool.bindElements($('.js-search-tool'));
+        }
+
+        if (
```

Comment:
I'm not even sure, what this code achieves: once the weather renders and we are in a phablet breakpoint, we scroll the window to the top? Why would we do that?